ProcessLib/ComponentTransport/CreateComponentTransportProcess.cpp

```diff
@@ -140,13 +142,49 @@ std::unique_ptr<Process> createComponentTransportProcess(
     {
         std::vector<std::reference_wrapper<ProcessLib::ProcessVariable>>
             per_process_variable;
-
-        for (auto& pv : collected_process_variables)
+        if (!chemical_solver_interface)
         {
-            per_process_variable.emplace_back(pv);
-            process_variables.push_back(std::move(per_process_variable));
+            for (auto& pv : collected_process_variables)
+            {
+                per_process_variable.emplace_back(pv);
+                process_variables.push_back(std::move(per_process_variable));
+            }
         }
+        else
         {
+            auto const components =
+                chemical_solver_interface->getComponentList();
+            // pressure
+            per_process_variable.emplace_back(collected_process_variables[0]);
+            process_variables.push_back(std::move(per_process_variable));
+            // concentration
+            assert(components.size() + 1 == collected_process_variables.size());
+            std::transform(components.begin(), components.end(),
+                           std::back_inserter(process_variables),
+                           [&per_process_variable,
```

Comment:
style: move the lambda one level up placing it before the transform call. Makes it readable.

Reply:
Fixed.